examples/external-data/src/mock-customer-service/service.ts

```diff
@@ -12,29 +12,64 @@ import fetch from "node-fetch";
 import { ClientManager } from "../utilities";
 import { assertValidTaskData, ITaskData } from "../model-interface";
 
+/**
+ * Expected shape of the "broadcast-signal" message that is sent to the /broadcast-signal service endpoint.
+ */
+export interface BroadcastSignalBodyInterface {
+	/**
+	 * Content of signal. Required by server.
+	 */
+	signalContent: {
+		/**
+		 * Required parameter. Required by server. User may add more parameters than the required `type` and `content` below.
+		 */
+		contents: {
+			/**
+			 * Required parameter. User defined content. User defined content that will be pass
+			 * unchanged to client.
+			 */
+			content: JSON | string;
+			/**
+			 * Required parameter. User defined content that will be pass unchanged to client.
+			 * User may consider using this type to differentiate between multiple diffrent signal types,
+			 * and possibly use it for versioning as well.
+			 */
+			type: string;
+		};
+	};
+}
+
 /**
  * Submits notifications of changes to Fluid Service.
  */
 function echoExternalDataWebhookToFluid(
 	taskData: ITaskData,
 	fluidServiceUrl: string,
-	containerUrl: string,
 	externalTaskListId: string,
+	tenantId: string,
+	documentId: string,
 ): void {
+	const fluidService = `${fluidServiceUrl}/${tenantId}/${documentId}/broadcast-signal`;
 	console.log(
-		`CUSTOMER SERVICE: External data has been updated. Notifying Fluid Service at ${fluidServiceUrl}`,
+		`CUSTOMER SERVICE: External data has been updated. Notifying Fluid Service at ${fluidService}`,
 	);
 
-	// TODO: we will need to add details (like ContainerId) to the message body or the url,
-	// so this message body format will evolve
-	const messageBody = JSON.stringify({ taskData, containerUrl, externalTaskListId });
-	fetch(fluidServiceUrl, {
+	const messageBody = {
```

Comment:
I think adding the type here should work and we can remove the casts. If it doesn't work, this object doesn't conform to the interface.
```suggestion
	const messageBody: BroadcastSignalBodyInterface = {
```

Reply:
When I do this, I get this error:
<img width="1178" alt="Screenshot 2023-10-18 at 8 23 28 AM" src="https://github.com/microsoft/FluidFramework/assets/6777404/30370918-1a5c-4cdc-b9f9-d70e0facf6b4">

Reply:
This works though -- 
```
	const messageBody: BroadcastSignalBodyInterface = {
		signalContent: {
			contents: {
				content: JSON.stringify({ taskData, externalTaskListId }),
				type: "ExternalDataChanged_V1.0.0",
			},
		},
	};
```

It looks like turning the ITaskData interface into a JSON isn't happening without some extra handling, but stringifying it works fine?

Reply:
Ok, I think I misunderstood how that `JSON` type works and I'm not even finding official documentation about it 🤔 . We probably want to just type `contents.content` as `unknown`.

Reply:
Ohh I forgot I defined this type, I can simply change the type definitions for content! Silly me.